src/interactivity/base/InteractivityFactory.cpp

```diff
@@ -29,6 +29,88 @@
 using namespace std;
 using namespace Microsoft::Console::Interactivity;
 
+HRESULT PseudoConsoleWindowAccessibilityProvider::RuntimeClassInitialize(HWND pseudoConsoleHwnd) noexcept
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, pseudoConsoleHwnd);
+    _pseudoConsoleHwnd = pseudoConsoleHwnd;
+    return S_OK;
+}
+
+IFACEMETHODIMP PseudoConsoleWindowAccessibilityProvider::get_ProviderOptions(_Out_ ProviderOptions* pOptions)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, pOptions);
+    *pOptions = ProviderOptions_ServerSideProvider;
+    return S_OK;
+}
+
+IFACEMETHODIMP PseudoConsoleWindowAccessibilityProvider::GetPatternProvider(_In_ PATTERNID /*iid*/,
+                                                                            _COM_Outptr_result_maybenull_ IUnknown** ppInterface)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, ppInterface);
+    *ppInterface = nullptr;
+    return S_OK;
+}
+
+IFACEMETHODIMP PseudoConsoleWindowAccessibilityProvider::GetPropertyValue(_In_ PROPERTYID propertyId,
+                                                                          _Out_ VARIANT* pVariant)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, pVariant);
+
+    pVariant->vt = VT_EMPTY;
+
+    // Returning the default will leave the property as the default
+    // so we only really need to touch it for the properties we want to implement
+    if (propertyId == UIA_ControlTypePropertyId)
+    {
+        pVariant->vt = VT_I4;
+        pVariant->lVal = UIA_WindowControlTypeId;
+    }
+    else if (propertyId == UIA_NamePropertyId)
+    {
+        pVariant->bstrVal = SysAllocString(AutomationPropertyName);
+        if (pVariant->bstrVal != nullptr)
+        {
+            pVariant->vt = VT_BSTR;
+        }
+    }
+    else if (propertyId == UIA_IsControlElementPropertyId)
+    {
+        pVariant->vt = VT_BOOL;
+        pVariant->boolVal = VARIANT_FALSE;
+    }
+    else if (propertyId == UIA_IsContentElementPropertyId)

```

Comment:
would you mind doing this in a `switch...case`? It will look a lot better when you collapse all the false boolean cases into one body as well...